controller/admin_controller.php

```diff
@@ -0,0 +1,296 @@
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->language = $language;
+		$this->log = $log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->phpbb_root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * Display the options a user can configure for this extension
+	 *
+	 * @return null
+	 * @access public
+	 */
+	public function display_options()
+	{
+		$this->template->assign_vars(array(
+			'IDEAS_POSTER'		=> $this->get_ideas_topics_poster_username(),
+			'IDEAS_BASE_URL'	=> ($this->config['ideas_base_url']) ?: '',
+
+			'S_FORUM_SELECT_BOX'	=> $this->select_ideas_forum(),
+			'S_IDEAS_FORUM_ID'		=> !empty($this->config['ideas_forum_id']),
+
+			'U_ACTION'			=> $this->u_action,
+			'U_FIND_USERNAME'	=> append_sid("{$this->phpbb_root_path}memberlist.{$this->php_ext}", 'mode=searchuser&amp;form=acp_phpbb_ideas_settings&amp;field=ideas_poster_id&amp;select_single=true'),
+		));
+	}
+
+	/**
+	 * Set configuration options
+	 *
+	 * @return null
+	 * @access public
+	 */
+	public function set_config_options()
+	{
+		$errors = array();
+
+		// This method is called on submit, so set flag to true initially
+		$submit = true;
+
+		$this->cfg_array = $this->request->variable('config', array('' => ''), true);
+
+		// Check the form for validity
+		if (!check_form_key('acp_phpbb_ideas_settings'))
+		{
+			$errors[] = $this->language->lang('FORM_INVALID');
+		}
+
+		// Check if Ideas forum is set
+		if (empty($this->cfg_array['ideas_forum_id']))
+		{
+			$errors[] = $this->language->lang('ACP_IDEAS_NO_FORUM');
+		}
+
+		// Check if selected user exists
+		$user_id = $this->get_ideas_topics_poster_id();
+		if (!$user_id)
+		{
+			$errors[] = $this->language->lang('NO_USER');
+		}
+
+		// Don't save settings if errors have occured
+		if (sizeof($errors))
+		{
+			$submit = false;
+		}
+
+		if ($submit)
+		{
+			// If selected user does exist, reassign the config value to its ID
+			$this->cfg_array['ideas_poster_id'] = $user_id;
+
+			// Configuration options to list through
+			$display_vars = array(
+				'ideas_forum_id',
+				'ideas_poster_id',
+				'ideas_base_url',
+				'ideas_forum_setup',
+			);
+
+			// We go through the display_vars to make sure no one is trying to set variables he/she is not allowed to
+			foreach ($display_vars as $config_name)
+			{
+				if (!isset($this->cfg_array[$config_name]))
+				{
+					continue;
+				}
+
+				if ($submit)
+				{
+					$this->config->set($config_name, $this->cfg_array[$config_name]);
+				}
+			}
+			$this->log->add('admin', $this->user->data['user_id'], $this->user->ip, 'ACP_PHPBB_IDEAS_SETTINGS_LOG');
+			trigger_error($this->language->lang('ACP_IDEAS_SETTINGS_UPDATED') . adm_back_link($this->u_action));
+		}
+
+		$this->template->assign_vars(array(
+			'S_ERROR'	=> (bool) sizeof($errors),
+			'ERROR_MSG'	=> (sizeof($errors)) ? implode('<br />', $errors) : '',
+		));
```

Comment:
If you want, I think you can move a simpler version of this up into the if (seizeof(errors)) at line 136 like:
```php
$this->template->assign_vars(array(
	'S_ERROR'	=> true,
	'ERROR_MSG'	=> implode('<br />', $errors),
));
```